csharp/ql/lib/ext/Microsoft.AspNetCore.Components.model.yml

```diff
@@ -5,6 +5,9 @@ extensions:
     data:
       - ["Microsoft.AspNetCore.Components", "NavigationManager", True, "get_BaseUri", "", "", "ReturnValue", "remote", "manual"]
       - ["Microsoft.AspNetCore.Components", "NavigationManager", True, "get_Uri", "", "", "ReturnValue", "remote", "manual"]
+      - ["Microsoft.AspNetCore.Components", "ParameterAttribute", False, "", "", "Attribute.Getter", "ReturnValue", "remote", "manual"]
```

Comment:
Can properties with `[Parameter]` attribute be controlled by the user? Aren't these only component parameters that are required for passing data from component to component?

Reply:
So I double checked:

> Can properties with [Parameter] attribute be controlled by the user? 

Yes, routing parameters for page components which are potentially controlled by a user navigating to those pages.

Example:

```razor
@page "/people-search/{year:int}/{name}"

@code {
   [Parameter]
   public string Name { get; set; }
   
   [Parameter]
   public int Year { get; set; }
}
```

> Aren't these only component parameters that are required for passing data from component to component?

So while they aren't "only ... for passing data from component to component", that is the primary use case for them.

Because of that I have decided to remove `[Parameter]` from this PR and I will add some additional logic in QL to reason about it better.